[Adjusted to apply to Debian's 2.6.26]

commit 4c31e92b97b6d7e7b19ee5e54a22571ffdebb305
Author: Yinghai Lu <yinghai@kernel.org>
Date:   Wed Apr 22 14:19:27 2009 -0700

    x86: check boundary in setup_node_bootmem()
    
    Commit dc09855 ("x86/uv: fix init of memory-less nodes") causes a
    two sockets system (where node-1 doesn't have RAM installed) to crash.
    
    That commit makes node_possible include cpu nodes that do not have memory.
    So check boundary in setup_node_bootmem().
    
    [ Impact: fix boot crash on RAM-less NUMA node system ]
    
    Signed-off-by: Yinghai Lu <yinghai@kernel.org>
    Cc: Jack Steiner <steiner@sgi.com>
    LKML-Reference: <49EF89DF.9090404@kernel.org>
    Signed-off-by: Ingo Molnar <mingo@elte.hu>

diff -urpN a/arch/x86/mm/numa_64.c b/arch/x86/mm/numa_64.c
--- a/arch/x86/mm/numa_64.c	2010-03-10 23:58:05.000000000 -0700
+++ b/arch/x86/mm/numa_64.c	2010-03-10 23:56:43.000000000 -0700
@@ -198,6 +198,9 @@ void __init setup_node_bootmem(int nodei
 	const int pgdat_size = round_up(sizeof(pg_data_t), PAGE_SIZE);
 	int nid;
 
+	if (!end)
+		return;
+
 	start = round_up(start, ZONE_ALIGN);
 
 	printk(KERN_INFO "Bootmem setup node %d %016lx-%016lx\n", nodeid,
